src/kzg/mod.rs

```diff
@@ -305,14 +305,22 @@ fn eval_poly(blob: &[Fr], z: &Fr) -> Fr {
 }
 
 fn compute_challenge(blob: &[Fr], commitment: &G1Affine) -> Fr {
-    let mut data = String::from("FSBLOBVERIFY_V1_");
-    let degree_separator: [u8; 16] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 10, 0];
-    data.push_str(std::str::from_utf8(&degree_separator).unwrap());
-    let encoded_blob = bincode::serialize(&blob).expect("should be able to serialize blob");
-    data.push_str(&format!("{:x?}", encoded_blob));
-    let encoded_commitment =
-        bincode::serialize(&commitment).expect("should be able to serialize commitment");
-    data.push_str(&format!("{:x?}", encoded_commitment));
+    let mut data = String::from("FSBLOBVERIFY_V1_").into_bytes();
+    let degree_separator: [u8; 16] = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0x10, 0];
+    data.extend(&degree_separator);
+    let blob_bytes = blob
+        .iter()
+        .flat_map(|el| {
+            el.into_repr()
+                .0
+                .iter()
+                .rev()
+                .flat_map(|v| v.to_be_bytes())
+                .collect::<Vec<u8>>()
```

Comment:
you need a length assert here, and did they really decide to make a commitment based on the full blob bytes? This is strange move